### PR TITLE
BHBC-993: Review access requests + misc user/role management endpoints + misc cleanup

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-993

## This PR contains the following changes

Add Review Access Requests component
- Modified the manage users page and child components so that all pre-render api calls happen in the parent manage users component.  This way we can trigger refreshes for both lists when a request is reviewed (same as the project view page).

Lots of backend endpoints added.

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [x] Misc cleanup / Refactoring / Documentation
- [ ] Version change

### CODE REVIEW

File: app/src/pages/access/AccessRequestPage.test.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const mockUseBiohubApi = {
   codes: {
     getAllCodeSets: jest.fn<Promise<object>, []>()
   },
-  accessRequest: {
+  admin: {
```
Comment (NickPhura) [inline app/src/pages/access/AccessRequestPage.test.tsx:17]: Ahhh, right.  Good catch

File: api/src/paths/access-request.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+    if (!identitySource) {
+      throw new HTTP400('Missing required body param: identitySource');
+    }
+
+    if (!administrativeActivityId) {
+      throw new HTTP400('Missing required body param: requestId');
+    }
+
+    if (!administrativeActivityStatusTypeId) {
```
Comment (sdevalapurkar) [inline api/src/paths/access-request.ts:119]: question:
do we not need to check for `!roleIds`?

File: api/src/queries/administrative-activity/administrative-activity-queries.ts [hunk 3]
```diff
@@ -147,3 +150,46 @@ export const countPendingAdministrativeActivitiesSQL = (userIdentifier: string):
 
   return sqlStatement;
 };
+
+/**
+ * SQL query update and existing administrative activity record.
```
Comment (sdevalapurkar) [inline api/src/queries/administrative-activity/administrative-activity-queries.ts]:
```suggestion
 * SQL query update an existing administrative activity record.
```

File: app/src/components/dialog/RequestDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  onDeny: () => void;
+  /**
+   * Callback fired if the 'Approve' button is clicked.
+   *
+   * @memberof IRequestDialog
+   */
+  onApprove: (values: any) => void;
+}
+
+const RequestDialog: React.FC<IRequestDialog> = (props) => {
```
Comment (sdevalapurkar) [inline app/src/components/dialog/RequestDialog.tsx:60]: this file needs tests (not necessarily in this pr)

File: app/src/features/admin/users/AccessRequestList.tsx [hunk 5]
```diff
@@ -47,103 +56,179 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+    request: IGetAccessRequestsListResponse | any;
+  }>({
+    open: false,
+    request: null
+  });
 
-  useEffect(() => {
-    const getAccessRequests = async () => {
-      const accessRequestResponse = await biohubApi.admin.getAccessRequests();
+  const [openErrorDialogProps, setOpenErrorDialogProps] = useState<IErrorDialogProps>({
```
Comment (sdevalapurkar) [inline app/src/features/admin/users/AccessRequestList.tsx:93]: this file needs tests (not required in this pr)

File: app/src/hooks/useKeycloakWrapper.test.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { cleanup } from '@testing-library/react';
```
Comment (sdevalapurkar) [inline app/src/hooks/useKeycloakWrapper.test.tsx]: what is this file for.... seems incomplete and possibly in the wrong spot

File: api/src/paths/access-request.ts (hunk 2 above)
Comment (NickPhura) [reply api/src/paths/access-request.ts:119]: No, as this endpoint is generic to both approved and denied updates.  So in the case of a Denied request, we wouldn't pass any roles.

File: app/src/hooks/useKeycloakWrapper.test.tsx (hunk 6 above)
Comment (NickPhura) [reply app/src/hooks/useKeycloakWrapper.test.tsx]: Not sure haha, will remove

File: app/src/components/dialog/RequestDialog.tsx (hunk 4 above)
Comment (NickPhura) [reply app/src/components/dialog/RequestDialog.tsx:60]: Started on it, but will save for a separate PR

**Review by anissa-agahchen — APPROVED** [review]
Looks good.

**Review by sdevalapurkar — APPROVED** [review]

**Pull request merged**